packages/api/src/functions/dbAuth/DbAuthHandler.ts

```diff
@@ -382,30 +403,30 @@ export class DbAuthHandler {
   // creates and returns a user, first checking that the username/password
   // values pass validation
   async _createUser() {
-    const { username, password, ...userAttributes } = JSON.parse(
-      this.event.body as string
-    )
-    this._validateField('username', username)
-    this._validateField('password', password)
+    const { username, password, ...userAttributes } = this.params
+    if (
+      this._validateField('username', username) &&
+      this._validateField('password', password)
+    ) {
+      const user = await this.dbAccessor.findUnique({
+        where: { [this.options.authFields.username]: username },
+      })
+      if (user) {
+        throw new DbAuthError.DuplicateUsernameError(username as string)
+      }
 
-    const user = await this.dbAccessor.findUnique({
-      where: { [this.options.authFields.username]: username },
-    })
-    if (user) {
-      throw new DbAuthError.DuplicateUsernameError(username)
+      // if we get here everything is good, call the app's signup handler and let
+      // them worry about scrubbing data and saving to the DB
+      const [hashedPassword, salt] = this._hashPassword(password as string)
+      const newUser = await this.options.signupHandler({
+        username,
```

Comment:
@dac09 This one is still red squiggles...also shows in Gitpod, but I'm already working on the code locally so I don't know that it actually does me any good to open it there.